examples/qwen2.5_nemo/training/model/train.py

```diff
@@ -0,0 +1,187 @@
+import argparse
+import os
+import nemo_run as run
+import lightning.pytorch as pl
+from nemo.collections import llm
+
+import torch
+print(f"CUDA available: {torch.cuda.is_available()}")
+print(f"Number of GPUs: {torch.cuda.device_count()}")
+
+### Dataset
+from data import BespokeDataModule
```

Comment:
is there a reason why we need to define this custom data class instead of relying on the pre-defined huffingface dataset one?